src/Document_Item.php

```diff
@@ -326,84 +327,57 @@ public static function showForDocument(Document $doc)
-                'checkright'
-                                                      => true
-            ]);
-            echo "</td><td class='center'>";
-            echo "<input type='submit' name='add' value=\"" . _sx('button', 'Add') . "\" class='btn btn-primary'>";
-            echo "<input type='hidden' name='documents_id' value='$instID'>";
-            echo "</td></tr>";
-            echo "</table>";
-            Html::closeForm();
-            echo "</div>";
-        }
-
-        echo "<div class='spaced table-responsive'>";
-        if ($canedit && $number) {
-            Html::openMassiveActionsForm('mass' . __CLASS__ . $rand);
-            $massiveactionparams = ['container' => 'mass' . __CLASS__ . $rand];
-            Html::showMassiveActions($massiveactionparams);
-        }
-        echo "<table class='tab_cadre_fixehov'>";
-
-        $header_begin  = "<tr>";
-        $header_top    = '';
-        $header_bottom = '';
-        $header_end    = '';
-
-        if ($canedit && $number) {
-            $header_top    .= "<th width='10'>" . Html::getCheckAllAsCheckbox('mass' . __CLASS__ . $rand);
-            $header_top    .= "</th>";
-            $header_bottom .= "<th width='10'>" . Html::getCheckAllAsCheckbox('mass' . __CLASS__ . $rand);
-            $header_bottom .= "</th>";
-        }
-
-        $header_end .= "<th>" . _n('Type', 'Types', 1) . "</th>";
-        $header_end .= "<th>" . __('Name') . "</th>";
-        $header_end .= "<th>" . Entity::getTypeName(1) . "</th>";
-        $header_end .= "<th>" . __('Serial number') . "</th>";
-        $header_end .= "<th>" . __('Inventory number') . "</th>";
-        $header_end .= "</tr>";
-        echo $header_begin . $header_top . $header_end;
-
+            $twig_params = [
+                'doc' => $doc,
+                'add_item_msg' => __('Add an item'),
+                'add_btn_msg' => _x('button', 'Add'),
+            ];
+            // language=Twig
+            echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
+                {% import 'components/form/fields_macros.html.twig' as fields %}
+                {% import 'components/form/basic_inputs_macros.html.twig' as inputs %}
+                {% set rand = random() %}
+                <div class="mb-3">
+                    <form name="documentitem_form{{ rand }}" id="documentitem_form{{ rand }}" method="post"
+                          action="{{ 'Document_Item'|itemtype_form_path }}">
+                        {{ inputs.hidden('_glpi_csrf_token', csrf_token()) }}
+                        {{ inputs.hidden('documents_id', doc.fields['id']) }}
+                        {{ fields.dropdownItemsFromItemtypes('', add_item_msg, {
+                            'itemtypes': doc.getItemtypesThatCanHave(),
+                            'entity_restrict': doc.fields['is_recursive'] ? doc.getSonsOf('glpi_entities', doc.fields['entities_id']) : doc.fields['entities_id'],
```

Comment:
`doc.getSonsOf()` method does not exists, maybe `entity_restrict` should be passed as a template variable.